tools/codegen/api/types.py

```diff
@@ -22,6 +22,10 @@
 class BaseCppType:
     ns: Optional[str]
     name: str
+    # This is only applicable to C++ reference types, like IntArrayRef and OptionalScalarRef.
+    # If defined, it describes a function that takes in the name of a variable of the given type,
+    # and emits code to generate a copy.
+    copy_ref_type: Optional[Callable[[str], str]] = None
```

Comment:
Callable in the field type seems a little smelly to me (in an OO language, one would always ask, then, why this isn't just a method)

Reply:
(woops forgot to delete, this is going away)